### PR TITLE
DON-298 update comms pref sections

### PR DESCRIPTION
**Pull request by tbg-jasonfung**
- DON-298 dynamically define opt in copy from campaign API + accommodate for backwards compatibility
- Update unit test campaign model structure

### CODE REVIEW

**Comment by tbg-jasonfung** [comment]
Merging in favour of Full sandbox refresh.

File: src/app/analytics.service.spec.ts [hunk 1]
```diff
@@ -47,6 +47,17 @@ describe('AnalyticsService', () => {
         regulatorNumber: '123456',
         regulatorRegion: 'Scotland',
       },
+      {
+        charityOptIn: 'Charity opt in',
```
Comment (NoelLH) [inline src/app/analytics.service.spec.ts:51]: @tbg-jasonfung what's going on with all of these?!

Comment (tbg-jasonfung) [reply src/app/analytics.service.spec.ts:51]: @NoelLH it's custom labels I mentioned [here](https://thebiggive.slack.com/archives/DCJLDGN2J/p1603095146002400)

Comment (NoelLH) [reply src/app/analytics.service.spec.ts:51]: @tbg-jasonfung these are still not documented per-value anywhere I can see, and it doesn't seem like the examples are realistic so the tests don't help at all in understanding the purposes.

Comment (tbg-jasonfung) [reply src/app/analytics.service.spec.ts:51]: @NoelLH this might help paint a better picture: https://github.com/thebiggive/donate-frontend/pull/332/files

**Pull request merged**